### PR TITLE
feat: [CDS-56687]: Improve backward compatibility of Rancher k8s tasks

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
## Describe your changes
Improve compatibility of rancher k8s tasks with older delegates. This change would ensure a distinct helpful message appears in case of older delegates.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49305)

### CODE REVIEW

File: 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java [hunk 1]
```diff
@@ -488,7 +488,8 @@ public enum TaskType {
   RANCHER_LIST_CLUSTERS_TASK_NG(TaskGroup.K8S_NG, "Rancher list clusters task for NG"),
   COMMAND_TASK_NG_WITH_OUTPUT_VARIABLE_SECRETS(TaskGroup.COMMAND_TASK_NG, "Command Task"),
   NOTIFY_WEBHOOK(TaskGroup.NOTIFICATION),
-  SERVERLESS_ROLLBACK_V2_TASK(TaskGroup.SERVERLESS_NG, "Serverless Rollback V2 Task");
+  SERVERLESS_ROLLBACK_V2_TASK(TaskGroup.SERVERLESS_NG, "Serverless Rollback V2 Task"),
+  K8S_COMMAND_TASK_NG_RANCHER(TaskGroup.K8S_NG, "K8s Task for NG Rancher deployments");
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java:492]: We should mark it unsupported until we aren't done with the feature itself

Comment (abhinavsingh1196) [reply 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java:492]: sure, will mark it

**Review by abhinavsingh1196 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49305)* status: 7 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java` line 177 at r1](https://reviewable.io/reviews/harness/harness-core/49305#-NXytVLWBkSkIRqRSb4Z:-NXzRM-n59cFQB9d3Cei:b-37l6yg) ([raw file](https://github.com/harness/harness-core/blob/2916f550e4a4736e38454e1aaebabaf61da8270d/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java#L177)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

Are you sure this is a good logic? What if you want to use service hooks together with a rancher?  The first "IF" will check that FF it's enabled and check if service hooks exist and it will return K8S_COMMAND_TASK_NG_V2. It won't consider the rancher part at all. I think you would just need to switch places of those two "IF". First, check if it's Rancher and then if it has service hooks.
</blockquote></details>

rearranged these and wrote some tests to preserve the order.

**Review by alebuh — APPROVED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49305)* status: 7 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by abhinavsingh1196** [comment]
trigger review

**Review by abosii-harness — APPROVED** [review]

**Pull request merged**